File: Codes/Hash/Non-Cryptographic/APHash/APHash.cs
/*
	Hash function by Arash Partow
    Archive of Reversing.ID
    Non-Cryptographic Hash
*/

// Fungsi ini merupakan method dari sebuah class
public static uint APHash (string key)
{
    uint state = 0xAAAAAAAA;
    uint i = 0;

    for (i = 0; i < key.Length; i++)
    {
        state ^= ((i & 1) == 0) ? ( (state <<  7) ^ ((byte)key[(int)i]) *  (state >> 3)) : 
		                         (~((state << 11) + (((byte)key[(int)i]) ^ (state >> 5))));
    }
    return state;
}

File: Codes/Hash/Non-Cryptographic/APHash/APHash.java
/*
	Hash function by Arash Partow
    Archive of Reversing.ID
    Non-Cryptographic Hash
*/

// Fungsi ini merupakan method dari sebuah class
public static long APHash (String key)
{
    long state = 0xAAAAAAAA;

    for (long i = 0; i < key.length(); i++)
    {
        if ((i & 1) == 0)
        {
            state ^=   ((state << 7) ^ key.charAt(i) *  (state >> 3));
        }
        else
        {
            state ^= (~((state << 11) + key.charAt(i) ^ (state >> 5)));
        }
    }
    return state;
}